### PR TITLE
Username Field

### PR DESCRIPTION
**Pull request by adeelshahid**

### CODE REVIEW

File: src/core_modules/capture-core/src/utils/validators/form/username.validator.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * @param {string} value
+ * @param {boolean} [isEmptyValid=false] By default empty is invalid. In your form you should precheck this and not call isValidUrl if empty.
+ * @returns
+ */
+export default function isValidUsername(value: string, isEmptyValid: boolean = false) {
+  if (!value) {
+    return isEmptyValid;
+  }
+
+  return value.length > 3
```
[the remark this answers is not in the document]
Comment (adeelshahid) [reply src/core_modules/capture-core/src/utils/validators/form/username.validator.js]: just an common assumption what's your suggestion

**Comment by adeelshahid** [comment]
@JoakimSM username field uses material-ui theme set by default. I have added _this.props.onBlur_ on handleChange event.

username-ui

**Comment by Bekkalizer** [comment]
Added the username to the end of a data entry field, and this expands the section:
skjermbilde 2018-07-26 kl 21 04 07

**Comment by Bekkalizer** [comment]
Some comments:
-The username type fields needs to display the data elements displayFormName - this is the context under which the user selects a user name.
-When selecting a username and saving, the username is sent and stored correctly in the back end database. However, the already saved username is not shown in the data entry form when reopening the event.

**Comment by adeelshahid** [comment]
@Bekkalizer

- The username type fields needs to display the data elements displayFormName - this is the context under which the user selects a user name.

Any place I can view the form field displayed

- When selecting a username and saving, the username is sent and stored correctly in the back end database. However, the already saved username is not shown in the data entry form when reopening the event.

Can you give me the endpoints for GET/POST where to

- save the usernames
- get the list of usernames selected previously

**Comment by adeelshahid** [comment]
I am closing this PR, opening a new one with code fixes in one place

**Pull request closed without merging**